ccm_web/client/src/App.tsx

```diff
@@ -1,18 +1,61 @@
 import React from 'react'
-import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
+import { BrowserRouter as Router, Link as RouterLink, Route, Switch } from 'react-router-dom'
+import { Breadcrumbs, Link, makeStyles, Typography } from '@material-ui/core'
 
 import ConsumerTest from './components/ConsumerTest'
-import Home, { mergeSectionProps } from './pages/Home'
-import MergeSections from './pages/MergeSections'
+import Home, { featureCardProps } from './pages/Home'
 import './App.css'
 
+const useStyles = makeStyles((theme) => ({
+  breadcrumbs: {
+    paddingLeft: 25,
+    paddingTop: 25
+  }
+}))
+
 function App (): JSX.Element {
+  const classes = useStyles()
   return (
     <div className='App'>
       <Router>
+        <div className={classes.breadcrumbs}>
+          <Route>
+              {({ location }) => {
+                const pathnames = location.pathname.split('/').filter(x => x)
+                return (
+                  <Breadcrumbs aria-label="breadcrumb">
+                    <Link component={RouterLink} to='/'>
+                      <Typography color='textPrimary'>
+                        Canvas Course Manager
+                      </Typography>
+                    </Link>
+
+                    {pathnames.map((value, index) => {
```

Comment:
@chrisrrowland  These seems to be variables used below and I can see types are missing. should have been `(value: string, index: int)`. 

Trying to understand why is my IDE is complaining type are missing. We do have strict mode enabled for type. why aren't these flagged.

I am trying to understand how the types linting is working here. Same is true for line 24 where `const pathnames = location.pathname.split('/').filter(x => x) ` can be expressed as
`const pathnames: string = location.pathname.split('/').filter(x => x)`

@ssciolla you can also give some input

Reply:
I'm not sure.  I have seen warnings before when it says the type can be inferred and is not necessary.  
In this case I don't get a warning if the types are explicitly specified or left out.
I'm using ESLint 2.1.19 in VSCode

Reply:
What is the lint error code? @pushyamig

Reply:
Currently, I don't see any errors with regards to why types are missing.

Reply:
Ok, I and Chris had conversation certain time types are inferred.